src/main/groovy/io/seqera/wave/service/metric/impl/MetricsServiceImpl.groovy

```diff
@@ -0,0 +1,156 @@
+import io.micronaut.cache.annotation.Cacheable
+import io.micronaut.core.annotation.Nullable
+import io.seqera.wave.service.license.LicenseManClient
+import io.seqera.wave.service.metric.MetricConstants
+import io.seqera.wave.service.metric.MetricsCounterStore
+import io.seqera.wave.service.metric.MetricsService
+import jakarta.inject.Inject
+import jakarta.inject.Singleton
+/**
+ * Implements service to retrieve wave metrics
+ *
+ * @author Munish Chouhan <munish.chouhan@seqera.io>
+ */
+@Slf4j
+@Singleton
+@CompileStatic
+class MetricsServiceImpl implements MetricsService {
+
+    @Inject
+    private MetricsCounterStore metricsCounterStore
+
+    @Inject
+    @Nullable
+    private LicenseManClient licenseManager
+
+    DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd")
+
+    @Override
+    Long getBuildsMetrics(String date, String org) {
+        return metricsCounterStore.get(getBuildsKey(date, org)) ?: 0
+    }
+
+    @Override
+    Long getPullsMetrics(String date, String org) {
+        return metricsCounterStore.get(getPullsKey(date, org)) ?: 0
+    }
+
+    @Override
+    Long getFusionPullsMetrics(String date, String org) {
+        return metricsCounterStore.get(getFusionPullsKey(date, org)) ?: 0
+    }
+
+    @Override
+    void incrementFusionPullsCounter(String token){
+        def org = getOrg(token)
+        def key = getFusionPullsKey(LocalDate.now().format(dateFormatter), null)
+            metricsCounterStore.inc(key)
+        log.trace("increment Fusion Pulls Count: $key")
+        if( org ){
+            key = getFusionPullsKey(null, org)
+            metricsCounterStore.inc(key)
+            key = getFusionPullsKey(LocalDate.now().format(dateFormatter), org)
+            metricsCounterStore.inc(key)
+        }
+    }
+
+    @Override
+    void incrementBuildsCounter(String token){
+        def org = getOrg(token)
+        def key = getBuildsKey(LocalDate.now().format(dateFormatter), null)
+        metricsCounterStore.inc(key)
+        log.trace("increment Builds Count: $key")
+        if( org ) {
+            key = getBuildsKey(null, org)
+            metricsCounterStore.inc(key)
+            key = getBuildsKey(LocalDate.now().format(dateFormatter), org)
+            metricsCounterStore.inc(key)
+        }
+    }
+
+    @Override
+    void incrementPullsCounter(String token) {
+        def org = getOrg(token)
+        def key = getPullsKey(LocalDate.now().format(dateFormatter), null)
+        metricsCounterStore.inc(key)
+        log.trace("increment Pulls Count: $key")
+        if ( org ) {
+            key = getPullsKey(null, org)
+            metricsCounterStore.inc(key)
+            key = getPullsKey(LocalDate.now().format(dateFormatter), org)
+            metricsCounterStore.inc(key)
+        }
+    }
+
+    @Cacheable('cache-1min')
+    protected String getOrg(String token) {
+        return licenseManager && token ? licenseManager.checkToken(token, "tower-enterprise").organization : null
```

Comment:
Let's infer the org from the email host instead, I mean if the user is `paolo@seqera.io` then the org is `seqera.io`

Reply:
ok I will make the changes

Reply:
done